dcos-log/api/v2/handlers.go

```diff
@@ -347,22 +382,22 @@ func discover(w http.ResponseWriter, req *http.Request) {
 
 	if err != nil {
 		errMsg := fmt.Sprintf("unable to get canonical task ID: %s", err)
-		logError(w, errMsg, http.StatusInternalServerError)
+		logError(w, req, errMsg, http.StatusInternalServerError)
 		return
 	}
 
-	taskURL, err := redirectURL(canonicalTaskID, file, req.URL.RawQuery)
+	taskURL, err := redirectURL(canonicalTaskID, file, req.URL.RawQuery, browse, download)
 	if err != nil {
 		errMsg := fmt.Sprintf("unable to build redirect URL: %s", err)
-		logError(w, errMsg, http.StatusInternalServerError)
+		logError(w, req, errMsg, http.StatusInternalServerError)
 		return
 	}
 
 	http.Redirect(w, req, taskURL, http.StatusSeeOther)
 }
 
-func journalHandler(w http.ResponseWriter, r *http.Request) {
-	acceptHeader := r.Header.Get("Accept")
+func journalHandler(w http.ResponseWriter, req *http.Request) {
```

Comment:
small nit: i feel like these changes where we're just renaming a parameter could be put into a separate commit to ease the burden on the reviewer. not a big deal tho.